### DISCUSSION TITLE
warn about decorations of pointer values in return type

### DISCUSSION THREAD

**Discussion by foldcat** [odin-lang/Odin#4483] [Ideas/Requests]
given this piece of code
```odin
package test

import "core:fmt"

build_foo :: proc() -> (result: ^[dynamic]int) {
	append(result, 1)
	return
}

main :: proc() {
	f := build_foo()
	fmt.println(f) // <nil>
}
```
will not function as the result is simply a pointer, but for me this isn't obvious at the first glance

I suggest emitting warnings when the compiler encounters this kind of code

**Comment by flysand7** [comment]
Not sure I completely understand. I'm struggling to see anything dangerous or weird about the example you have posted. Can you clarify what exactly there is to warn about?

**Reply by foldcat** [reply]
result is a pointer declared at the return section of the procedure `build_foo`
from my understanding it is only a pointer and not an actual dynamic array so no matter how you append to it it will still be nil

if you run the code nil will be printed but there were clearly an attempt of appending 1 to it

**Reply by flysand7** [reply]
Isn't `append(<nil>, {})` a runtime error?

**Comment by gingerBill** [comment]
The compiler cannot warn about a user-space level construct which could be perfectly valid.
